src/SprykerEco/Zed/Heidelpay/Business/Processor/Notification/Converter/NotificationXmlConverter.php

```diff
@@ -51,12 +51,18 @@ protected function loadXml(string $xml): SimpleXMLElement
      */
     protected function simpleXmlToArray(SimpleXMLElement $xmlElement): array
     {
-        $result = (array)$xmlElement;
+        $result = [];
 
-        foreach ($result as $name => $value) {
-            if ($value instanceof SimpleXMLElement) {
-                $result[$name] = $this->simpleXmlToArray($value);
-            }
+        $attributes = $xmlElement->attributes();
+        if ($xmlElement->count() > 0 && $attributes->count() > 0) {
+            $result['@attributes'] = ((array)$attributes)['@attributes'];
+        }
+
+        foreach ($xmlElement->children() as $node) {
+            /** @var \SimpleXMLElement $node */
+            $result[$node->getName()] = $node->count() > 0 ? $this->simpleXmlToArray($node) : (string)$node;
+
+            return $result;
```

Comment:
This is not correct. You do not want to return at this place. It looks like this is a copy-paste error from #52 . You should remove the return here.

Reply:
Thanks, I've fixed it. 
Yes, it's copypaste just to release the smallest and the fastest patch to not block integration.
We will prepare the fix later, for now, I think it's enough. 
I'll be right back with an answer when we can merge it.